lib/sport_ngin_aws_auditor/instance_helper.rb

```diff
@@ -42,7 +43,7 @@ def compare(tag_name)
       end
       
       add_instances_with_tag_to_hash(instances_with_tag, differences)
-      differences
+      return differences, retired_tags_hash
```

Comment:
Returning arrays of complex objects technically works fine, but the complexity makes for poor readability and brittle code.  Consider refactoring and encapsulating in a class.